### ISSUE TITLE
When running `setup.py clean` variables aren't defined

### ISSUE THREAD

**Issue by s7726** [Toblerity/Fiona#872]
The prior if is where these variables are defined. So in a 'clean' situation this fails.

https://github.com/Toblerity/Fiona/blob/c4dfadffeb60a37ceaa8d6a4b338f2f22779d1f5/setup.py#L174

```
  Running setup.py clean for fiona
  ERROR: Command errored out with exit status 1:
   command: 'C:\Users\[userdir]\covid_env\Scripts\python.exe' -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'C:\\Users\\[userdir]\\AppData\\Local\\Temp\\pip-install-je_yzyb9\\fiona\\setup.py'"'"'; __file__='"'"'C:\\Users\\[userdir]\\AppData\\Local\\Temp\\pip-install-je_yzyb9\\fiona\\setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' clean --all
       cwd: C:\Users\[userdir]\AppData\Local\Temp\pip-install-je_yzyb9\fiona
  Complete output (5 lines):
  Traceback (most recent call last):
    File "<string>", line 1, in <module>
    File "C:\Users\[userdir]\AppData\Local\Temp\pip-install-je_yzyb9\fiona\setup.py", line 186, in <module>
      log.info("GDAL version major=%r minor=%r", gdal_major_version, gdal_minor_version)
  NameError: name 'gdal_major_version' is not defined
  ----------------------------------------
  ERROR: Failed cleaning build dir for fiona

**Comment by s7726** [comment]
@sgillies do you wait for it to be in a build, rather than just merged to close it? Or do you expect the contributor (me) to close it? No worries, just wondering.

**Comment by sgillies** [comment]
@s7726 I'm leaving it open, it will be closed automatically once we merge the maint-1.8 branch with the master branch again.
